includes/Shopping/Woocommerce_Query.php

```diff
@@ -76,9 +75,23 @@ public function get_search( string $search_term ) {
 		_prime_post_caches( $products_image_ids, false, true );
 
 		foreach ( $products as $product ) {
-			$product_image_ids = array_map( 'absint', $product->get_gallery_image_ids() );
-			$images            = array_map( [ $this, 'get_product_images' ], $product_image_ids );
+			$product_image_ids = array_unique(
+				array_map(
+					'absint', 
+					array_merge(
+						[ $product->get_image_id() ],
+						$product->get_gallery_image_ids()
+					)
+				)
+			);
```

Comment:
I don't love this nested array_map, this is not very readable.

Reply:
```suggestion
			$product_image_ids = $product->get_gallery_image_ids();
			$product_image_ids[] = $product->get_image_id();
			$product_image_ids = array_map( 'absint', $product_image_ids );
			$product_image_ids = array_unique( array_filter( $product_image_ids ) );
```

This feels a little cleaner.

Reply:
This suggestion moves `$product->get_image_id()` to the end of the array, not the beginning. I think it's better to have the main product image to be at the start of the array.

Reply:
Agreed - fixed